lib/puma/launcher.rb

```diff
@@ -48,18 +48,21 @@ def initialize(conf, launcher_args={})
 
       env = launcher_args.delete(:env) || ENV
 
-      @config.options[:log_writer] = @log_writer
+      @config.load
+      @config.clamp
+      @options = @config.options
+
+      @options[:log_writer] = @log_writer
 
       # Advertise the Configuration
       Puma.cli_config = @config if defined?(Puma.cli_config)
+      log_config if env['PUMA_LOG_CONFIG']
```

Comment:
I've QA'ed and confirmed that moving this here doesn't change it's position in the actual logs.